integrations/tensorflow/e2e/keras/iree_vision_test_suite.bzl

```diff
@@ -59,59 +131,64 @@ def iree_vision_test_suite(
-                    )
-                    tests.append(test_name)
+    all_configurations_set = sets.make([
+        (model, dataset, backend)
+        for model in models
+        for dataset in datasets
+        for backend in backends
+    ])
 
-                    args = [
-                        "--model={}".format(model),
-                        "--data={}".format(dataset),
-                        "--include_top=1",
-                        "--target_backends={}".format(",".join(test_backends)),
-                    ]
-                    if external_weights:
-                        args.append("--url={}".format(external_weights))
+    failing_configurations_set = sets.make([])
+    if failing_configurations != None:
+        # Parse failing configurations.
+        for configuration in failing_configurations:
+            # Normalize configuration input.
+            for key, value in configuration.items():
+                if type(value) == type(""):
+                    configuration[key] = [value]
 
-                    # TODO(GH-2175): Simplify this after backend names are standardized.
-                    driver = backend.replace("iree_", "")  # "iree_<driver>" --> "<driver>"
-                    if driver == "llvmjit":
-                        driver = "llvm"
-                    py_test_tags = ["driver={}".format(driver)]
-                    if tags != None:  # `is` is not supported.
-                        py_test_tags += tags
+            for model in configuration["models"]:
+                for dataset in configuration["datasets"]:
+                    for backend in configuration["backends"]:
+                        sets.insert(
+                            failing_configurations_set,
+                            (model, dataset, backend),
+                        )
 
-                    iree_py_test(
-                        name = test_name,
-                        main = "vision_model_test.py",
-                        srcs = ["vision_model_test.py"],
-                        args = args,
-                        tags = py_test_tags,
-                        deps = deps,
-                        size = size,
-                        python_version = python_version,
-                        **kwargs
-                    )
+        # Generate failing targets.
+        failing_tags = [
+            "manual",
+            "nokokoro",
+            "notap",
+        ]
+        if tags != None:
+            failing_tags += tags
 
-    native.test_suite(
-        name = name,
-        tests = tests,
-        # Note that only the manual tag really has any effect here. Others are
-        # used for test suite filtering, but all tests are passed the same tags.
-        tags = tags,
-        # If there are kwargs that need to be passed here which only apply to
-        # the generated tests and not to test_suite, they should be extracted
-        # into separate named arguments.
+        _generate_test_suite(
```

Comment:
FYI there's another way to do this as well. test_suite filters based on the tags you give it, so you could add a "failing" tag to the failing tests and then create a test suite with that tag and one excluding that tag. https://docs.bazel.build/versions/master/be/general.html#test_suite.tags

So this could be a single iteration over model, dataset, and backend which includes a check that adds the appropriate tags if it's in the failing configurations and then adds the failing tags. Up to you whether you think that's better or it's worth changing.

Reply:
Thoughts? There are a few less lines this way.

Reply:
LGTM